src/widgets/call-panel.jsx

```diff
@@ -388,38 +433,74 @@ class CallPanel extends React.PureComponent {
     this.props.onHangup(this.props.topic, this.props.seq);
   }
 
-  // Mute/unmute audio/video (specified by kind).
-  toggleMedia(kind) {
+  // Ends video track and turns off the camera.
+  muteVideo() {
     const stream = this.state.localStream;
-    stream.getTracks().forEach(track => {
-      if (track.kind != kind) {
-        return;
-      }
-      track.enabled = !track.enabled;
-    });
-    // Make sure we redraw the mute/unmute icons (e.g. mic -> mic_off).
-    this.forceUpdate();
+    const t = stream.getVideoTracks()[0];
+    t.enabled = false;
+    t.stop();
+
+    stream.removeTrack(t);
+  }
+
+  unmuteVideo() {
+    const pc = this.state.pc;
+    navigator.mediaDevices.getUserMedia({ video: true })
+      .then(stream => {
+        // Will extract video track from stream and throw stream away,
+        // and replace video track in the media sender.
+        this.localRef.current.srcObject = null;
+        const sender = pc.getSenders().find(s => s.track.kind == 'video');
+        const track = stream.getVideoTracks()[0];
+        // Remote track from new stream.
+        stream.removeTrack(track);
+        // Add this track to the existing local stream.
+        this.state.localStream.addTrack(track);
+        return sender.replaceTrack(track);
+      })
+      .then(_ => {
+        this.localRef.current.srcObject = this.state.localStream;
+      })
+      .catch(this.handleGetUserMediaError)
+      .finally(() => { this.forceUpdate() }); // Make sure we redraw the mute/unmute icons (e.g. camera -> camera_off).
```

Comment:
Consistently `()` or `_`

`.finally(_ => this.forceUpdate());`

Reply:
Done.